genesis/engine/entities/rigid_entity/rigid_link.py

```diff
@@ -121,82 +122,88 @@ def _build(self):
         for vgeom in self._vgeoms:
             vgeom._build()
 
-        self._init_mesh = self._compose_init_mesh()
-
-        # inertial_mass and inertia_i
-        if self._inertial_mass is None:
-            if len(self._geoms) == 0 and len(self._vgeoms) == 0:
-                self._inertial_mass = 0.0
+        if self._inertial_mass is None or self._inertial_pos is None or self._inertial_i is None:
+            # Determine which geom list to use: geoms first, then vgeoms, then fallback
+            if len(self._geoms) > 0:
+                geom_list = self._geoms
+                get_mesh = lambda geom: trimesh.Trimesh(geom.init_verts, geom.init_faces)
+                get_pos = lambda geom: geom._init_pos
+                get_quat = lambda geom: geom._init_quat
+            elif len(self._vgeoms) > 0:
+                geom_list = self._vgeoms
+                get_mesh = lambda geom: trimesh.Trimesh(geom.init_vverts, geom.init_vfaces)
+                get_pos = lambda geom: geom._init_pos
+                get_quat = lambda geom: geom._init_quat
             else:
-                if self._init_mesh.is_watertight:
-                    self._inertial_mass = self._init_mesh.volume * self.entity.material.rho
-                else:  # TODO: handle non-watertight mesh
-                    self._inertial_mass = 1.0
+                # Fallback: use default values
```

Comment:
This is not a fallback. It is the right solution where there is no geometry attached to a link.